doc/reference/trd-appid.md

```diff
@@ -0,0 +1,651 @@
+  same application binary. The process binaries have no credentials.
+  The verifier policy assigns a local application identifier to each
+  process binary based on its order in application flash.
+
+An application identifier provides an identity for an application
+binary. It allows the Tock kernel to know about the provenance and
+origin of the binary and make access control or security decisions
+based on this information. For example, a kernel may allow only
+applications whose credentials use a particular trusted public key to
+access restricted functionality, but restrict other applications to
+use a subset of available system calls.
+
+Application identifiers are distinct from process
+identifiers. An application identifier is per-application (persists
+across restarts of a process binary, for example), while a process
+identifier identifies a particular execution of that binary. At any
+time on a Tock device, each process has a unique process identifier,
+but they can be re-used over time (like POSIX process identifiers).
+
+As the above examples illustrate, application credentials can vary in
+size and content. The credentials that a kernel's verifier policy will
+accept depends on its use case. Certain devices will only accept
+credentials which include a particular public key, while others will
+accept many. Furthermore, the internal format of these credentials can
+vary.  Finally, the cryptography used in credentials can vary, either
+due to security policies or certification requirements.
+
+Because the verifier policy is responsible for assigning application
+identifiers to process binaries, it is possible for the same process
+binary to have different application identifiers on different Tock
+systems.  For example, suppose a process binary has two application
+credential TBF headers: one signs with a key A, and the other with key
+B. Tock systems using a verifier policy that accepts key A may assign
+A as the global application identifier, while Tock systems using a
+different verifier policy that accepts key B may assign B as the
+global application identifier.
+
+4 Credentials in Tock Binary Format Objects
+===============================
+
+Application credentials are usually stored in a [Tock Binary
+Format][TBF] object, along with the process binary they are associated
+with. They are usually stored as footers (after the TBF header and
+application binary) to simplify computing integrity values such as
+checksums or hashes. This requires have a TBF header that specifies
+where the application binary ends and the footers begin, information
+which the previous `TbfHeaderV2Main` header (the Main Header) does not
+include.  Including application credentials in a process binary
+therefore requires using an alternative `TbfHeaderV2Program` header
+(the Program Header), which specifics where footers begin. This
+section describes the format and semantics of Program Headers and
+Credentials Footers.
+
+
+4.1 Program Header
```

Comment:
Why not just add a new main header version?

Reply:
The versions related to TBF versions, not header versions.